netctl/commands.go

```diff
@@ -168,27 +174,37 @@ var Commands = []cli.Command{
 				Name:      "ls",
 				Aliases:   []string{"list"},
 				Usage:     "List rules for a given tenant/policy",
-				ArgsUsage: "[policy]",
-				Flags:     []cli.Flag{tenantFlag, allFlag, jsonFlag},
-				Action:    listRules,
+				ArgsUsage: "",
+				Flags: []cli.Flag{tenantFlag, allFlag, jsonFlag, quietFlag,
+					cli.StringFlag{
+						Name:  "policy, p",
+						Usage: "Policy - REQUIRED",
+					},
+				},
+				Action: listRules,
 			},
 			{
 				Name:      "rm",
 				Aliases:   []string{"delete"},
 				Usage:     "Delete an existing rule.",
-				ArgsUsage: "[policy] [rule id]",
-				Flags:     []cli.Flag{tenantFlag},
-				Action:    deleteRule,
+				ArgsUsage: "[rule id]",
```

Comment:
@shaleman While everything that you said makes sense and I agree with it. But, from a UX perspective, even if we keep auto-completion aside for a moment; there is still a slight element of ambiguity/extra-responsibility that we enforce on the user by making the positioning mandatory. `netctl policy add-rule <policy-name> <rule-id>` the policy-name and rule-id should be entered in the exact same order. This is not very intuitive as the user will have to see help to get to it. so, In my opinion, either we should do `netctl rule add --policy <policy-name> <rule-id>` or we should do `netctl policy add-rule --policy <policy-name> <rule-id>`. That way, the user can hit tabs and get to what he would want.